### PR TITLE
feat: Added order by most popular in Onboarding screen 

### LINKED ISSUE #9858: [CAL-2077] Onboarding / Add conferencing - Order by most popular
We should order by most popular conferencing app first. We built this logic for ordering in the app store I believe so we can likely re-use that.

[2dc26491-bd98-41dd-a4b1-2dfefa8c2e51](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/21b63a24-c497-4c14-8b80-0cf40d365907/2dc26491-bd98-41dd-a4b1-2dfefa8c2e51)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2077](https://linear.app/calcom/issue/CAL-2077/onboarding-add-conferencing-order-by-most-popular)</sub>

### PR DESCRIPTION
**Pull request by JaideepGuntupalli**
## What does this PR do?

I have added the same sorting for the most popular apps used in App Store to the onboarding screen. First, I have separated the function into its file and then imported it to the integrations handler. Also added sortByMostPopular flag so it doesn't messup existing usages of the endpoint. I have only enabled the flag in Calendar and Conferencing onboarding screens.

Fixes #9858 

https://www.loom.com/share/588776ff1fd147a884eaa8307a68645d?sid=9009ab6c-c322-4564-96b6-a60850b853f5

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)

## How should this be tested?

- When testing locally, you need first to install a few apps, as the most popular apps are compared based on no. of installations it has.
- Then try to signup with a new account and compare with the following screenshot which is default when no apps are installed.

| Calendar | Conferencing |
|--------|--------|
| 10cfeaeb637fbcfb2abc16a733db8d97 | 7a3618dd1ee4b20f2772291b7e0bb8d4 |

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by PeerRich** [comment]
great iniative! this will help a lot. we will review this this week

**Comment by PeerRich** [comment]
CleanShot 2023-07-06 at 15 37 54@2x
seems to be some errors

**Comment by JaideepGuntupalli** [comment]
@PeerRich fixed the error, but getting another error from teamId which is also on the main branch too when i run `yarn run type-check:ci`
Screenshot 2023-07-07 at 10 42 36

File: packages/trpc/server/routers/loggedInViewer/integrations.handler.ts [hunk 1]
```diff
@@ -33,8 +34,16 @@ type TeamQuery = Prisma.TeamGetPayload<{
...
+  const {
+    variant,
+    exclude,
+    onlyInstalled,
+    includeTeamInstalledApps,
+    extendsFeature,
+    teamId,
+    sortByMostPopular,
+  } = input;
+  const { credentials } = user;
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/loggedInViewer/integrations.handler.ts:46]:
```suggestion
  let { credentials } = user;
```
causes type error, as we assign a new value to `credentials` later on

Comment (JaideepGuntupalli) [reply packages/trpc/server/routers/loggedInViewer/integrations.handler.ts]: Ok now looking back at it was a mistake from resolving conflicts. The ide showed the error but i couldn't find why, but it all makes sense now. Thanks for that @CarinaWolli !!

**Comment by JaideepGuntupalli** [comment]
Screenshot 2023-07-09 at 09 08 56
@PeerRich the issue is resolved now

**Comment by JaideepGuntupalli** [comment]
hey @PeerRich, just a reminder, can you pls review the PR

File: packages/lib/apps/getMostPopularApps.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { z } from "zod";
+
+import prisma from "@calcom/prisma";
+
+const getMostPopularApps = async () => {
```
Comment (hariombalhara) [inline packages/lib/apps/getMostPopularApps.ts:5]: I think this function doesn't give most popular apps nor it does any ordered result.
It just gives the count of installs per appId. We should rename this accordingly.

**Review by hariombalhara — APPROVED** [review]
LGTM!! Thanks again @JaideepGuntupalli.

If you find my comment appropriate you can open up a followup PR to do the renaming

**Pull request merged**